### ISSUE TITLE
Engine 1.55 no longer builds API docs

### ISSUE THREAD

**Issue by yaustar** [playcanvas/engine#4486]
Error message
```
ERROR: Unable to parse a tag's type expression for source file /Users/syau/Snapchat/Dev/playcanvas-public/engine/src/graphics/program-lib/programs/lit-shader.js in line 311 with tag title "type" and text "{[string, string, string, any[]]}": Invalid type expression "[string, string, string, any[]]": Expected "!", "$", "'", "(", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/syau/Snapchat/Dev/playcanvas-public/engine/src/graphics/program-lib/programs/lit-shader.js in line 312 with tag title "type" and text "{[string, string, string, any[]]}": Invalid type expression "[string, string, string, any[]]": Expected "!", "$", "'", "(", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/syau/Snapchat/Dev/playcanvas-public/engine/src/graphics/webgl/webgl-shader.js in line 288 with tag title "returns" and text "{[string, {message?: string, line?: number, source?: string}]} A tuple where the first element is the 10 lines of code around the first
detected error, and the second element an object storing the error messsage, line number and
complete shader source.": Invalid type expression "[string, {message?: string, line?: number, source?: string}]": Expected "!", "$", "'", "(", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
ERROR: Unable to parse a tag's type expression for source file /Users/syau/Snapchat/Dev/playcanvas-public/engine/src/graphics/webgl/webgl-shader.js in line 299 with tag title "returns" and text "{[string, {message?: string, line?: number, source?: string}]} A tuple where the first element is the 10 lines of code around the first
detected error, and the second element an object storing the error messsage, line number and
complete shader source.": Invalid type expression "[string, {message?: string, line?: number, source?: string}]": Expected "!", "$", "'", "(", "*", ".", "...", "0", "?", "@", "Function", "\"", "\\", "_", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "implements", "import", "in", "instanceof", "interface", "let", "new", "null", "package", "private", "protected", "public", "return", "static", "super", "switch", "this", "throw", "true", "try", "typeof", "undefined", "var", "void", "while", "with", "yield", "{", Unicode letter number, Unicode lowercase letter, Unicode modifier letter, Unicode other letter, Unicode titlecase letter, Unicode uppercase letter, or [1-9] but "[" found.
```

**Comment by kungfooman** [comment]
It's a bug in catharsis (whose author is aware of it since 2017 but too lazy to fix his JSDoc PEG grammar) so I fixed it here: https://github.com/playcanvas/engine/pull/4368

The affected files are not in docs and the rest of the doc files build fine.

**Comment by slimbuck** [comment]
Fixed in #4493
